TombEngine/Game/effects/effects.cpp

```diff
@@ -1302,17 +1302,35 @@ void TriggerFireFlame(int x, int y, int z, FlameType type)
 		}
 		else
 		{
-			spark->sR = 255;
-			spark->sB = 48;
-			spark->sG = (GetRandomControl() & 0x1F) + 48;
+			if (type == FlameType::Color || type == FlameType::ColorTrail)
```

Comment:
What I would do here is to check if `color1` argument is provided at all, and then switch to colored mode. One may want to use custom color with any flame type (if this makes sense at all).

Reply:
If Lara hits a normal flame emitter while burning with color, color1 will not update so I made a new FlameType. In the future I plan that each flame emitter can have a color, as well as the burning torch.